draft-barnes-acme.md

```diff
@@ -198,324 +212,362 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
 
 | Code            | Semantic                                                 |
 |:================|:=========================================================|
 | malformed       | The request message was malformed                        |
 | unauthorized    | The client lacks sufficient authorization                |
 | serverInternal  | The server experienced an internal error                 |
-| notSupported    | The request type is not supported                        |
-| unknown         | The server does not recognize an ID/token in the request |
 | badCSR          | The CSR is unacceptable (e.g., due to a short key)       |
 
-The server may also defer providing a response by sending a defer message.  For example, in the key authorization process, the server may need additional time to validate the client's responses to its challenges.  Or in the issuance process, there may be some delay due to batch signing.  The fields in a defer message are as follows:
+Authorization and challenge objects can also contain error information to indicate why the server was unable to validate authorization.
 
-type (required, string):
-: "defer"
+TODO: Flesh out errors and syntax for them
 
-token (required, string):
-: An opaque value that the client uses to check on the status of the request (using a statusRequest message).
+## Registration
 
-interval (optional, number):
-: The amount of time, in seconds, that the client should wait before checking on the status of the request.  (This is a recommendation only, and clients SHOULD enforce minimum and maximum deferral times.)
+An ACME registration resource represents a set of metadata associated to an account key pair, most importantly contact information and a recovery token.  Registration resources have the following structure:
 
-message (optional, string):
-: A human-readable string describing the reason for the deferral.
+key (required, dictionary):
+: The public key of the account key pair, encoded as a JSON Web Key object {{I-D.ietf-jose-json-web-key}}.
 
-For example, a deferral due to batch signing might be indicated with a message of the following form:
+contact (optional, array of string):
+: An array of URIs that the server can use to contact the client for issues related to this authorization. For example, the server may wish to notify the client about server-initiated revocation, or check with the client on future authorizations (see the "recoveryContact" challenge type).
+
+recoveryToken (optional, string):
+: An opaque token that the client can present to demonstrate that it participated in a prior authorization transaction.
+
+A client creates a new account with the server by sending a POST request to the server's new-registration URI.  The body of the request is a registration object containing only the "contact" field.
```

Comment:
+ The server MUST reject requests containing "key" or "recoveryToken" fields. If new fields are added in the future, the server SHOULD explicitly enumerate which fields are copied into the server-side registration object.